CaImAn/caiman_parameters.py

```diff
@@ -0,0 +1,129 @@
+            "strides": (neuron_diameter[-1]*4, neuron_diameter[-1]*4),
+            "overlaps": (neuron_diameter[-1]*2, neuron_diameter[-1]*2),
+            "max_deviation_rigid": neuron_diameter[0]/2,
+            "border_nan": "copy",
+            "method_init": "corr_pnr",  # use this for 1 photon
+            "K": None,
+            "gSig": (neuron_diameter[0], neuron_diameter[0]),
+            "merge_thr": 0.8,
+            "p": 1,
+            "tsub": self.parameters[defs.Parameters.danse.TEMPORAL_DOWNSAMPLE],
+            "ssub": self.parameters[defs.Parameters.danse.SPATIAL_DOWNSAMPLE],
+            "rf": neuron_diameter[-1]*4,
+            "stride": neuron_diameter[-1]*2,
+            "only_init": True,    # set it to True to run CNMF-E
+            "nb": 0,
+            "nb_patch": 0,
+            "method_deconvolution": "oasis",       # could use "cvxpy" alternatively
+            "low_rank_background": None,
+            "update_background_components": True,  # sometimes setting to False improve the results
+            "min_corr": self.parameters[defs.Parameters.danse.LOCAL_CORR_THRESHOLD],
+            "min_pnr": self.parameters[defs.Parameters.danse.PNR_THRESHOLD],
+            "normalize_init": False,               # just leave as is
+            "center_psf": True,                    # leave as is for 1 photon
+            "ssub_B": 2,
+            "ring_size_factor": 1.4,
+            "del_duplicates": True,
+            "use_cnn": False,
+            "fnames": self.paths[defs.Parameters.Path.TMP_DIR] + '/' + f"tiff_{'_'.join(self.get_h5path_names()[2:4])}.tif"
+            }
+
+        print(cm_defs.Messages.WRITE_IMAGE_TIFF, flush=True)
+        imwrite(self.params_caiman["fnames"], images.transpose(2, 0, 1))
+        del images
+
+        self.opts = params.CNMFParams(params_dict = self.params_caiman)
+        opts_dict, advanced_settings = self.set_advanced_parameters(self.opts, self.parameters.get(defs.Parameters.danse.ADVANCED_SETTINGS, {}))
+
+        #Update parameters and Advanced Setting
+        self.opts.change_params(opts_dict)
+        self.parameters[defs.Parameters.danse.ADVANCED_SETTINGS] = advanced_settings.copy()
+
+
+    def get_h5path_names(self):
+
+        """
+        Split the path to dataset into relevant names
+        """
+
+        h5path_names = self.paths[defs.Parameters.Path.H5PATH].split('/')
+
+        data = h5path_names[0]
+        driver = h5path_names[1]
+        operation = h5path_names[2]
+        series = h5path_names[-2]
+        sensor = h5path_names[-1]
+
+        return [data, driver, operation, series, sensor]
+
+
+    def set_advanced_parameters(self, param, advanced_parameters):
+
+        """
+        input:
+        param: is a class CNMFParams
+        advanced_parameters: dict
+
+        ouput:
+        new param: is a class CNMFParams
+        new advanced_parameters: dict
+        """
+        param_dict = param.to_dict()
+        advan_param_keys_used = []
```

Comment:
Feels like the names are confusing, could you try to rework them? For example, I would try to avoid using very long names if the information is redundant `advan_param_keys_used` -> keys_used or `cnmf_keys` or `caiman_keys`